### DISCUSSION TITLE
4:22:26 is not working for me.

### DISCUSSION THREAD

**Discussion by RuiAlvesPT** [smartcontractkit/full-blockchain-solidity-course-py#1618] [Q&A] [answered]
Edit: reformulated the question to be more clear.

Hey,
This is my code so far:

```import solcx
from solcx import compile_standard
import json
from web3 import Web3
import os
from dotenv import load_dotenv

load_dotenv()

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

    solcx.install_solc("0.6.0")
    compiled_sol = compile_standard(
        {
            "language": "Solidity",
            "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
            "settings": {
                "outputSelection": {
                    "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
                }
            },
        },
        solc_version="0.6.0",
    )

with open("compiled_code.json", "w") as file:
    json.dump(compiled_sol, file)

# get bytecode
bytecode = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["evm"][
    "bytecode"
]["object"]

# get abi
abi = compiled_sol["contracts"]["SimpleStorage.sol"]["SimpleStorage"]["abi"]


# for conecting to ganache (my own 1 node blockchain)
w3 = Web3(Web3.HTTPProvider("HTTP://127.0.0.1:8545"))
chain_id = 1337
my_address = "0x352266D8E6fa907AfBd34042200177C85f6d0904"
private_key = os.getenv("PRIVATE_KEY1")
# print(private_key)


# create a contract in python
SimpleStorage = w3.eth.contract(abi=abi, bytecode=bytecode)
# get latest transaction
nonce = w3.eth.getTransactionCount(my_address)
transaction = SimpleStorage.constructor().buildTransaction(
    {
        "gasPrice": w3.eth.gas_price,
        "chainId": chain_id,
        "from": my_address,
        "nonce": nonce,
    }
)
signed_txn = w3.eth.account.sign_transaction(transaction, private_key=private_key)

# send the signed transaction
tx_hash = w3.eth.send_raw_transaction(signed_txn.rawTransaction)
tx_receipt = w3.eth.wait_for_transaction_receipt(tx_hash)

# working with the contract you need:
# Contract address
# Contract ABI

simple_storage = w3.eth.contract(address=tx_receipt.contractAddress, abi=abi)

store_transaction = simple_storage.functions.store(15).buildTransaction(
    {
        "gasPrice": w3.eth.gas_price,
        "chainId": chain_id,
        "from": my_address,
        "nonce": nonce + 1,
    }
)
signed_store_txn = w3.eth.account.sign_transaction(
    store_transaction, private_key=private_key
)
send_store_tx = w3.eth.send_raw_transaction(signed_store_txn.rawTransaction)
tx_receipt = w3.eth.wait_for_transaction_receipt(send_store_tx)

print(simple_storage.functions.retrieve().call())
```

If I run python deploy.py I get the answer  [15](https://imgur.com/a/ytU17sP) as expected.  But as soon as I close the ganache UI and [run](https://imgur.com/a/DvsVrf8) ganache on powershell and changing my address, port and private key to new ones I then try to run python deploy.py and [this](https://user-images.githubusercontent.com/97696041/182036923-6ebd099e-3aec-4f93-8f63-fda9975d20d8.png) happens. Why? I already installed it before since it was giving me the 15... Thanks.

**Comment by cromewar** [comment, marked as answer]
Hello @RuiAlvesPT 
We will need more info on this, pelase give a check to #980 to reformulate the question.
Anyways you are saying that you cannot deploy de contract unless you have Ganache UI opened because the CLI is not working for you?

**Reply by RuiAlvesPT** [reply]
I am sorry, didnt knew the template needed, already edited the main post and yes that is kinda what I am saying, the ganache cli works, it creates me several addresses with ether and the private keys but the deploy dont work. Thanks

**Reply by cromewar** [reply]
Ok I've checked the code, and the error, did some research and it makes no sense 😂 as your windows profile is freaking out about a missing dependency you actually have, so this is my recommendations for you.

- create an isolated python instance with `virtualenv`
- install the packages on the new instance `py-solc-x` and so on.
- try again, does it work?

If keeps failing, please consider to migrate your workflow to WSL which is a linux instance running inside your windows machine.
Cheers.

**Reply by RuiAlvesPT** [reply]
Hi there, just to let you know that it worked now with the venv. What does that mean exactly for someone new like me? do I need to run this venv every time I want to program python now? Thank you so much!!

**Reply by cromewar** [reply]
Well I would suggest to have a venv for each type of project you are going to work with, lets say you are going to work with web3 projects and also machine learning projects, I would have 2 virtual envs, one for each type of project.
And yes, remember to activate the specific venv when you want to work with that project, but remember you don't need a different venv for each project, just for each Type of project.

Cheers.
